crates/fluvio-version-manager/src/command/switch/mod.rs

```diff
@@ -0,0 +1,94 @@
+//! Version Switching Command
+//!
+//! The `switch` command is responsible of changing the active Fluvio Version
+
+mod version_directory;
+
+use anyhow::Result;
+use clap::Parser;
+use colored::Colorize;
+
+use fluvio_hub_util::fvm::Channel;
+
+use crate::GlobalOptions;
+use crate::common::settings::Settings;
+use crate::common::workdir::fvm_versions_path;
+use crate::common::notify::Notify;
+
+use self::version_directory::VersionDirectory;
+
+#[derive(Debug, Parser)]
+pub struct SwitchOpt {
+    #[command(flatten)]
+    global_opts: GlobalOptions,
+    /// Version to set as active
+    #[arg(index = 1)]
+    version: Option<Channel>,
+}
+
+impl SwitchOpt {
+    pub async fn process(&self) -> Result<()> {
+        let Some(version) = &self.version else {
+            self.notify_help(format!(
+                "You can use {} to see installed versions",
+                "fvm show".bold()
+            ));
+
+            return Err(anyhow::anyhow!("No version provided"));
+        };
+
+        // Ensure the `~/.fvm/versions` directory exists given that we get
+        // installed binaries from there. Without this directory we cant
+        // switch versions.
+        let versions_path = fvm_versions_path()?;
+
+        if !versions_path.exists() {
+            self.notify_warn("No local Fluvio versions found.");
+            self.notify_help(format!(
+                "Try installing a version with {}, and then retry this command.",
+                "fvm install".bold()
+            ));
+
+            return Ok(());
+        }
+
+        // Build the path to the version directory requested by the user
+        // e.g. Version: 0.10.13 -> ~/.fvm/versions/0.10.13
+        let pkgset_path = versions_path.join(version.to_string());
+
+        if !pkgset_path.exists() {
+            self.notify_warn(format!(
+                "Fluvio version {} is not installed",
+                version.to_string().bold()
+            ));
+
+            let help = format!("fvm install {}", version);
+
+            self.notify_help(format!(
+                "Install the desired version using {}, and then retry this command.",
+                help.bold()
+            ));
+
+            return Ok(());
+        }
+
+        let version_dir = VersionDirectory::open(pkgset_path)?;
+
+        version_dir.set_active()?;
+
+        Settings::open()?.update_from_manifest(&version_dir.manifest)?;
```

Comment:
Why manifest files should be changed? This should be removed

Reply:
Manifest file will not change. What this does is updating the `~/.fvm/settings.toml` file to point to the active Fluvio Version.

Here Im using the `manifest` from the Version Directory as a source to update `settings.toml`.

Reply:
then VersionManager should update file since it is responsible for updating.